ostd/src/arch/x86/boot/smp.rs

```diff
@@ -107,6 +108,24 @@ fn init_boot_stack_array() {
     }
 }
 
+fn fill_boot_pt_ptr() {
+    // This is defined in the boot assembly code.
+    extern "C" {
+        fn __boot_page_table_pointer();
```

Comment:
I am very confused by this declaration. It is a pointer declared as a function. I expect the Rust declaration to be as follows.

```rust
extern "C" {
    static __boot_page_table_pointer: *const u8;
}
```

The `__ap_boot_stack_array_pointer` has exactly the same problem.